tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -524,7 +526,14 @@ export class WebGPUBackend extends KernelBackend {
         this.device, bindGroupLayout, inputs.map(t => this.tensorToBinding(t)),
         this.tensorToBinding(output), uniforms);
 
-    const encoder = this.device.createCommandEncoder();
+    let encoder: GPUCommandEncoder;
+    if (this.commandQueue.length > 0) {
```

Comment:
nit: Does this mean `commandQueue` is now always 0 or 1 in size? Could it just be `currentCommandEncoder: GPUCommandEncoder | undefined`?

Reply:
>Does this mean commandQueue is now always 0 or 1 in size?

Yes, I think it is.

>Could it just be currentCommandEncoder: GPUCommandEncoder | undefined?

Sorry, I am not fully understand your comment. Do you want me to change it like below:
```
let currentCommandEncoder: GPUCommandEncoder | undefined = this.commandQueue[0];
if (!currentCommandEncoder)
{
      currentCommandEncoder = this.device.createCommandEncoder();
      this.commandQueue.push(encoder);
}

Reply:
What I mean is, replace the `commandQueue: GPUCommandEncoder[]` with a `currentCommandEncoder: GPUCommandEncoder | undefined`, which is a simpler way to store 0-or-1 of something.

Anyway, not a blocking issue. Sorry for the delay in replying.

Reply:
Done